File: SYP.Web/Modules/ServerTypes/Setting/CountryColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CountryRow } from "./CountryRow";

export interface CountryColumns {
    Id: Column<CountryRow>;
    Name: Column<CountryRow>;
    Code: Column<CountryRow>;
    IsoCode3: Column<CountryRow>;
    PhoneCode: Column<CountryRow>;
    CountryNr: Column<CountryRow>;
    InsertUserId: Column<CountryRow>;
    InsertDate: Column<CountryRow>;
    UpdateUserId: Column<CountryRow>;
    UpdateDate: Column<CountryRow>;
}

export class CountryColumns extends ColumnsBase<CountryRow> {
    static readonly columnsKey = 'Setting.Country';
    static readonly Fields = fieldsProxy<CountryColumns>();
}

File: SYP.Web/Modules/Setting/Country/CountryColumns.cs
namespace SYP.Setting.Columns;

[ColumnsScript("Setting.Country")]
[BasedOnRow(typeof(CountryRow), CheckNames = true)]
public class CountryColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Code { get; set; }
    public string IsoCode3 { get; set; }
    public string PhoneCode { get; set; }
    public string CountryNr { get; set; }
    public int InsertUserId { get; set; }
    public DateTime InsertDate { get; set; }
    public int UpdateUserId { get; set; }
    public DateTime UpdateDate { get; set; }
}